Rectangle.cpp

```diff
@@ -1,15 +1,42 @@
 #ifndef Rectangle_cpp
 #define Rectangle_cpp
+
 #include <iostream>
-#include "Draw.cpp"
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
 
-class Rectangle : public Draw
+class Rectangle : public Shape , public Input
 {
 public:
     void drawShape() override
     {
         cout << "rectangle" << endl;
     }
+    int inputLength() override 
+    {
+        int length;
+        cout << "Enter Length" << endl;
+        cin >> length;
+        return length;
+    }
+    int inputWidth() override 
+    {
+        int width;
+        cout << "Enter Width" << endl;
+        cin >> width;
+         return width;
+    }
+    int CalculateArea() override
+    {
+        int length = inputLength ();
```

Comment:
Same as earlier, we are expanding the functional scope of the method `CalculateArea` by doing additional tasks on it.